workflow/controller/controller.go

```diff
@@ -1081,10 +1115,16 @@ func (wfc *WorkflowController) addWorkflowInformerHandlers(ctx context.Context)
 		},
 		Handler: cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
-				wfc.archiveWorkflow(ctx, obj)
+				key, err := cache.MetaNamespaceKeyFunc(obj)
+				if err == nil {
+					wfc.archiveWfQueue.Add(key)
```

Comment:
LGTM! But why call the `wfc.archiveWfQueue.Add` method instead of the `wfc.archiveWfQueue.AddRateLimited` method?

Reply:
I think it's not necessary to call AddRateLimited，the workflow_ttl_queue doesn't call AddRateLimited either.

Reply:
It is not necessary, but it does mean that this is not rate limited. It feels like this is a secondary function of the controller and would be better if it was rate limited. Is there a reason you choose not to rate limit it?

Reply:
> Is there a reason you choose not to rate limit it?

my read was that it was primarily a copy per above, and not for a particular reason

> the workflow_ttl_queue doesn't call AddRateLimited either.

Reply:
> It is not necessary, but it does mean that this is not rate limited. It feels like this is a secondary function of the controller and would be better if it was rate limited. Is there a reason you choose not to rate limit it?

the main reason is I want to keep the same with workflow_ttl_queue, because that two queues are similar, both of them are dealing with completed workflow, and that is not very frequent action IMO.